sys/arm64/arm64/machdep.c

```diff
@@ -903,6 +943,53 @@ sendsig(sig_t catcher, ksiginfo_t *ksi, sigset_t *mask)
 	tf = td->td_frame;
 	onstack = sigonstack(tf->tf_sp);
 
+#if __has_feature(capabilities)
+	/*
+	 * CHERI affects signal delivery in the following ways:
+	 *
+	 * (1) Additional capability state is exposed via extensions
+	 *     to the context frame placed on the stack.
+	 *
+	 * (2) If the user $pcc doesn't include CHERI_PERM_SYSCALL,
+	 *     then we consider user state to be 'sandboxed'.
+	 *
+	 * (3) If an alternative signal stack is not defined, and we
+	 *     are in a 'sandboxed' state, then we will terminate the
+	 *     process unconditionally.
+	 */
+	cheri_is_sandboxed = cheri_signal_sandboxed(td);
+
+	/*
+	 * We provide the ability to drop into the debugger in two different
+	 * circumstances: (1) if the code running is sandboxed; and (2) if the
+	 * fault is a CHERI protection fault.  Handle both here for the
+	 * non-unwind case.  Do this before we rewrite any general-purpose or
+	 * capability register state for the thread.
+	 */
+#ifdef DDB
+	if (cheri_is_sandboxed && security_cheri_debugger_on_sandbox_signal)
```

Comment:
We should probably axe this code from all architectures as a follow-up, we have GDB for userspace debugging.

Reply:
I already have a change to axe all the cheri_is_sandboxed bits from signal handling on aarch64 and RISC-V sitting in a branch off of this I'll post once this PR gets merged.